src/panels/MissionsPanel/MissionsPanel.tsx

```diff
@@ -1,13 +1,41 @@
-import { Stack, Text, ThemeIcon, Title } from '@mantine/core';
+import { Box, Select, Stack, Text, ThemeIcon, Title } from '@mantine/core';
 
+import { useOpenSpaceApi } from '@/api/hooks';
 import { RocketLaunchIcon } from '@/icons/icons';
-import { useAppSelector } from '@/redux/hooks';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import { setSelectedMission } from '@/redux/missions/missionsSlice';
 
 import { MissionContent } from './MissionContent';
 
 export function MissionsPanel() {
-  const hasMission = useAppSelector((state) => state.missions.isInitialized);
-  const mission = useAppSelector((state) => state.missions.data.missions[0]);
+  const {
+    isInitialized: hasMission,
+    data,
+    selectedMissionIdentifier
+  } = useAppSelector((state) => state.missions);
+
+  const luaApi = useOpenSpaceApi();
+  const dispatch = useAppDispatch();
+
+  let [mission] = data.missions;
```

Comment:
Personally I would prefer using this type of restructuring for hooks 
```suggestion
  let mission = data.missions.at(0);
```

Reply:
at first I did `data.missions[0]` but the linter screamed at me, I know we've done this in several other places so we should be consistent if we want to do the `at(0)` version